src/Mod/Arch/ArchWall.py

```diff
@@ -944,12 +949,16 @@ def getExtrusionData(self,obj):
                                 else:
                                     dvec.multiply(width/2)			## TODO width Value should be of no use (width/2), width Direction remains 'in use'
 
-                                    overrideWidthHalfen = [i/2 for i in obj.OverrideWidth]
+                                    overrideWidthHalfen = [i/2 for i in widths]
+
                                     w1 = DraftGeomUtils.offsetWire(wire,dvec,False, False, overrideWidthHalfen)
                                     dvec = dvec.negative()
                                     w2 = DraftGeomUtils.offsetWire(wire,dvec,False, False, overrideWidthHalfen)
 
                                 sh = DraftGeomUtils.bind(w1,w2)
+
+                            del widths[0:edgeNum]
```

Comment:
Isn't this deleting the entire widths content? Isn't it better to simply `del widths` or `widths = []`? Might be a bit faster

Reply:
Hmm, not exactly, see if i can explain better below:-

The whole list can contain width for every 'segment'/edge of wire in the sketch / wall.

For every 'loop' in 896 
`for i,wire in enumerate(self.basewires):`
... a 'continuous' series of edges / i.e. a wire is processed.

Then, 960 only take out from the list of "widths" the numbers of width figures corresponding to the numbers of edges in the wire.

Reply:
see also 898
edgeNum = len(wire.Edges)

then 960
del widths[0:edgeNum]